tests/scripts/check_test_cases.py

```diff
@@ -99,17 +101,11 @@ def walk_test_suite(self, data_file_name):
     def walk_ssl_opt_sh(self, file_name):
         """Iterate over the test cases in ssl-opt.sh or a file with a similar format."""
         descriptions = self.new_per_file_state() # pylint: disable=assignment-from-none
-        with open(file_name, 'rb') as file_contents:
-            for line_number, line in enumerate(file_contents, 1):
-                # Assume that all run_test calls have the same simple form
-                # with the test description entirely on the same line as the
-                # function name.
-                m = re.match(br'\s*run_test\s+"((?:[^\\"]|\\.)*)"', line)
-                if not m:
-                    continue
-                description = m.group(1)
-                self.process_test_case(descriptions,
-                                       file_name, line_number, description)
+        listed = subprocess.run(['tests/ssl-opt.sh', '-l'],
+                                 capture_output=True)
+        listed = set(map(lambda x: x.rstrip(), listed.stdout.splitlines()))
```

Comment:
Always use one of the `subprocess.check_xxx` functions unless you expect a nonzero status code (and even then, it might be better to catch the option). Here, if `ssh-opt.sh` fails, this isn't caught.

Reply:
Done